### PR TITLE
Require --no-node-snapshot in node 20+

### PR DESCRIPTION
**Pull request by hudsonb**
## Hey, I just made a Pull Request!

When started using node 20+ checks if the --no-node-snapshot option was provided and throws an error if not.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by benjdlambert — APPROVED** [review]
Nice :tada: Thanks for this! :pray:

File: .github/workflows/verify_e2e-windows.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
 
     env:
       CI: true
-      NODE_OPTIONS: --max-old-space-size=8192
+      NODE_OPTIONS: ${{ matrix.node-version == '20.x' && '--max-old-space-size=4096 --no-node-snapshot' || '--max-old-space-size=4096' }}
```
Comment (acierto) [inline .github/workflows/verify_e2e-windows.yml:29]: I'd keep the same value, not decreasing it.

Comment (hudsonb) [reply .github/workflows/verify_e2e-windows.yml]: Restored to 8192 in E2E tests

**Review by acierto — APPROVED** [review]

**Review by Rugvip — APPROVED** [review]
:shipit:

**Pull request merged**